neet/boolean/sensitivity.py

```diff
@@ -484,3 +488,160 @@ def average_sensitivity(self, states=None, weights=None, calc_trans=True):
                                            calc_trans=calc_trans)
 
         return np.sum(Q) / self.size
+
+    def C_sensitivity_at_x(self, state, transitions=None, c=1):
+        """C-Sensitivity modification of the regular sensitivity function. I deleted the 
+        doctest code because it was cluttering my screen
+
+        The c-sensitivity of f(x1, . . ., xn) at x is defined as the number of 
+        c-Hamming neighbors of x on which the function value is different from its value on x. That is,
+
+        :param state: a single network state
+        :type state: list, numpy.ndarray
+        :param transitions: precomputed state transitions (*optional*)
+        :type transitions: list, numpy.ndarray, None
+        :return: the C-sensitivity at the provided state
+
+        """
+
+        encoder = self._unsafe_encode
+        distance = self.distance
+        #neighbors = self.hamming_neighbors(state)
+        state_copy = copy.copy(state)
+        nextState = self.update(state)
+
+        """
+        Returns an iterator for each vector I which is a strict subset of {1,...,n} and where |I| = c
+        note: if c = 0, this will return an empty tuple
+        """
+
+        I_comb_iter = itt.combinations(range(self.size), c)
+
+        """ 
+        Generator function which returns a new hamming neighbor
+        Each hamming neighbor is simply the product of self.state XOR I
+        """
+        def c_hamming_neighbors(self, state, c):
+            try:
+                nxt = next(I_comb_iter)
+                XORed = copy.copy(state_copy)
+                for i in nxt:
+                    XORed[i] ^= 1
+                return XORed
+            except StopIteration:
+                return None
+
+
+        """ 
+        #OK, so I messed with the function and it's only ~kinda~ a generator function now...
+        It's automatically advanced in the for loop, which
+        acts as a "try: next(neighbors); catch StopIteration:". This behavior is built 
+        into Python and is idiomatic.
+        """ 
```

Comment:
BooleanNetwork.hamming_neighbors only works for 1-hamming neighbors since it creates a 1-bit-flipped copy of the network just by running through the indexes 0 to size-1. If I tried to apply BN.hamming_neighbors to the each network created by the original call to BN.hamming_neighbors I would end up with one copy in each 2nd-order call to BN.h_n which was equivalent to the original network. I could simply change BN.h_n to be my implementation though, which would allow me to call it, but I wanted to quarantine my test implementations while I was only starting out.